components/tidb_query/src/codec/row/v2.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{i16, i32, i8, u16, u32, u8};
+
+pub const CODEC_VERSION: u8 = 128;
+const BIG_ID_FLAG: u8 = 1;
+
+mod row_slice;
+
+#[cfg(test)]
+mod encoder;
+
+trait BoundedU64 {
```

Comment:
Maybe this is what you want: https://rust-num.github.io/num/num/trait.Bounded.html.
And we could use a blanket implementation for the BoundedU64 with another trait: https://rust-num.github.io/num/num/cast/trait.NumCast.html.

Reply:
the `bounded` value is used as Range patterns later,  so the value should either be literal or const value. So I think `num-trait` could not help.

I'm thinking maybe  using constant directly is more reasonable
```
mod int_bound {
    pub const MAX_U8: u64 = (1 << 8) - 1;
    pub const MAX_I8: i64 = (1 << 7) - 1;
    .......
}
```